tools/SourceKit/tools/sourcekitd/lib/API/VariableTypeArray.cpp

```diff
@@ -0,0 +1,177 @@
+//===---------- VariableTypeArray.cpp -------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "sourcekitd/VariableTypeArray.h"
+#include "sourcekitd/CompactArray.h"
+#include "SourceKit/Core/LLVM.h"
+#include "SourceKit/Core/LangSupport.h"
+#include "SourceKit/Support/UIdent.h"
+#include "DictionaryKeys.h"
+
+#include "llvm/Support/MemoryBuffer.h"
+
+using namespace SourceKit;
+using namespace sourcekitd;
+
+namespace {
+class VariableTypeReader {
+  const char *PrintedType;
+  // Four unsigned integers for:
+  //  - Variable offset in the source buffer
+  //  - Variable length in the source buffer
+  //  - Offset of printed Variable type inside printedType
```

Comment:
Nitpick: The `printedType` variable is capitalized: `printedType` -> `PrintedType`